### PR TITLE
CR for Worklist #20447 - Prepare Interface to include DDE face tracking

### PR DESCRIPTION
**Pull request by ctrlaltdavid**
Prior to DDE being included in the installer, you can install this build and then copy the DDE files into a subdirectory, \Interface\dde.

Operation:
- Developer > Avatar > Face Tracking > DDE Face Regression to turn on / off
- Developer > Avatar > Face Tracking > Reset DDE Tracking to reset tracking, or use Ctrl+''

Windows DDE runs, and I believe OSX DDE should run though I haven't tested it.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: ce104e8c4005ee6963c8051c237106f4c7446003 available at: https://gist.github.com/f8ef72c88f62f9e21c10<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4602.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4602.dmg

File: interface/src/devices/DdeFaceTracker.cpp [hunk 1]
```diff
@@ -19,8 +20,16 @@
 #include "DdeFaceTracker.h"
 #include "FaceshiftConstants.h"
 
-static const QHostAddress DDE_FEATURE_POINT_SERVER_ADDR("127.0.0.1");
-static const quint16 DDE_FEATURE_POINT_SERVER_PORT = 5555;
+#if defined(Q_OS_WIN)
+static const QString DDE_PROGRAM_PATH = QCoreApplication::applicationDirPath() + "/dde/dde.exe";
+static const QString DDE_ARGUMENTS = "--udp=127.0.0.1:64204 --receiver=64205 --headless";
+#elif defined(Q_OS_MAC)
+static const QString DDE_PROGRAM_PATH = QCoreApplication::applicationDirPath() + "/dde/dde";
```
Comment (murillodigital) [inline interface/src/devices/DdeFaceTracker.cpp]: @ctrlaltdavid can you switch this line to this:
static const QString DDE_PROGRAM_PATH = QCoreApplication::applicationDirPath() + "/dde.app/Contents/MacOS/dde";

We need to include the full dde app bundle

File: interface/src/devices/DdeFaceTracker.cpp [hunk 2]
```diff
@@ -19,8 +20,16 @@
...
+static const QString DDE_ARGUMENTS = "--udp=127.0.0.1:64204 --receiver=64205 --headless";
```
Comment (murillodigital) [inline interface/src/devices/DdeFaceTracker.cpp]: shouldn't we define DDE_(SERVER|CONTROL)_PORT above DDE_ARGUMENTS and use them in the ARGUMENTS string?

**Comment by hifi-gustavo** [comment]
Build console output for commit: f509d08378c62523b50e1132b3996b9463f3dd8b available at: https://gist.github.com/f8ef72c88f62f9e21c10<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4602.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4602.dmg

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: f509d08378c62523b50e1132b3996b9463f3dd8b available at: https://gist.github.com/f8ef72c88f62f9e21c10<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4602.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4602.dmg

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: f509d08378c62523b50e1132b3996b9463f3dd8b available at: https://gist.github.com/f8ef72c88f62f9e21c10<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4602.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4602.dmg

**Comment by murillodigital** [comment]
That last PR run already has DDE built into it --- all systems are ready for merging, release will include dde.

**Comment by OmegaHeron** [comment]
Some instructions might help - is this for the 2D face tracking or something else?

**Comment by OmegaHeron** [comment]
Tested on Windows 7.1 64 bit - works.  I won't comment on how well it works since what I've seen about it leads me to believe it's working as well as can be expected.  It tracks head movement, expression and eye blinks in general.

**Comment by nbq** [comment]
Tested on OSX 10.10.3 - DDE works but seems to not work as well for someone who is overweight or someone who wears glasses in my tests. This is the same problem I saw happen with FaceShift until the training so it leads me to assume it is tracking very well! It still caught eye blinks and mouth movements no matter what from my tests.

**Pull request merged**